### DISCUSSION TITLE
When the user clicks accept I need to refresh the page in order to see the scripts loading. Am i doing something wrong?

### DISCUSSION THREAD

**Discussion by nmenic01** [orestbida/cookieconsent#654] [Q&A]
So, my question is there a way to load the scripts without the need to refresh the page?

**Comment by orestbida** [comment]
Scripts should be enabled by the plugin automatically. You can test this with a simple example:

```html
<script type="text/plain" data-category="analytics">
    alert("accepted analytics");
</script>
```

As soon as you accept the analytics category, the script should be enabled and you should see the alert.

**Reply by nmenic01** [reply]
I tried doing that in a script:

```
<script type="text/plain" data-type="text/partytown" data-category="analytics">
    alert("accepted analytics");
</script>
```

But it won't work. When I remove the custom `data-type` it works fine though.

Any tips?

Thanks! :)

**Reply by orestbida** [reply]
I assume it doesn't work because partytown only fires on page load. When a category is enabled, you also need to notify partytown  about newly "added" scripts, as explained in their [docs](https://partytown.builder.io/partytown-scripts#dynamically-appending-scripts).

**Reply by orestbida** [reply]
One way to achieve it is to always notify partytown when something changes in cookieconsent:

```html
<script>
window.addEventListener('cc:onChange', () => {
   window.dispatchEvent(new CustomEvent('ptupdate'));
});
</script>
```

**Reply by nmenic01** [reply]
I tried `window.dispatchEvent(new CustomEvent('ptupdate'));` , even within the console and it won't fire again. 

I can see that the below works though:

```
window.addEventListener('cc:onConsent', () => {
   alert("triggered")
});
```

Not sure how to handle it at this point.

**Reply by orestbida** [reply]
Perhaps `alert` does not work as intended in partytown.

Try the following and check your browser's console. The script should be enabled and executed by PT after 3 seconds.
```html
<script type="text/plain" id="pt-test">
    console.log("dynamic pt works");
</script>

<script>
    setTimeout(() => {
        document.getElementById("pt-test").setAttribute("type", "text/partytown");
        window.dispatchEvent(new CustomEvent('ptupdate'));
    }, 3000);
</script>
```

If this does not work, there is an issue with your partytown setup.

**Reply by orestbida** [reply]
I personally tested cookieconsent + partytown and it works as expected. 

If you are still struggling, please create an example repo. so that I can take a look at your setup.

If this does not work:
```html
<script>
window.addEventListener('cc:onChange', () => {
   window.dispatchEvent(new CustomEvent('ptupdate'));
});
</script>
```
you can try dispatching the event from within the `onChange` callback:

```js
CookieConsent.run({
    ...
    onChange: () => {
        window.dispatchEvent(new CustomEvent('ptupdate'));
    }
});

**Reply by nmenic01** [reply]
I've tried your recommendations, but I couldn't make it work. Here's a sample of how my code looks like: 

https://stackblitz.com/edit/vitejs-vite-85mrff?file=src%2FPartyTownContainer.jsx

thank you!

**Reply by orestbida** [reply]
The demo you linked does not compile, looks like there are missing files (main.tsx)?

**Reply by orestbida** [reply]
I tried creating a simple Vite + React project and added CookieConsent v3 + Partytown and it works. Will share the demo later when I have a bit more time.

**Reply by nmenic01** [reply]
I re-added the missing pieces -> https://stackblitz.com/edit/vitejs-vite-85mrff?file=src%2FPartyTownContainer.jsx

Thanks a lot! 🙏🏼

**Reply by orestbida** [reply]
Alright, now the demo works.

I think I found the issue. You should add this:
```js
window.dispatchEvent(new CustomEvent('ptupdate'));
```

inside both the callback functions:
```js
CookieConsent.run({
    ...,
    onConsent: () => {
        window.dispatchEvent(new CustomEvent('ptupdate'));
    },
    onChange: () => {
        window.dispatchEvent(new CustomEvent('ptupdate'));
    }
});

**Reply by nmenic01** [reply]
Unfortunately, it still doesn't work for me.

Any chance you could share the demo, so I can have a look? 🙏🏼

**Reply by orestbida** [reply]
Here you go, a basic working demo https://github.com/orestbida/react-pt-cc
